multitool/private/templates.bzl

```diff
@@ -38,9 +40,40 @@ def _render_hub_tool(rctx, tool_name, filename, substitutions = None):
         } | (substitutions or {}),
     )
 
+def _render_tool_repo(hub_name, tool_name, binary):
+    name = "{name}.{tool_name}.{os}_{cpu}".format(
+        name = hub_name,
+        tool_name = tool_name,
+        binary = json.encode(binary),
+        os = binary["os"],
+        cpu = binary["cpu"],
+    )
+    return "\n".join([
+        "    tool_repo(",
+        "        name = \"{name}\",".format(name = name),
+        "        tool_name = \"{tool_name}\",".format(tool_name = tool_name),
+        "        binary = '{binary}',".format(binary = json.encode(binary)),
+        "    )",
+        "",
```

Comment:
there's probably a nicer way to do some of this templating but this will do for now